src/contexts/product-reducer.jsx

```diff
@@ -0,0 +1,33 @@
+export const productReducer = (state, action) => {
+    switch (action.type) {
+        case "SUCCESS": {
+            return {
+                ...state, ...state, products: action.data
+            };
+        }
+        case "ERROR": {
+            return {
+                ...state,
+                error: action.error,
+            };
+        }
+        case 'ADD_TO_CART':{
+            return {
+                ...state, cart: [...state.cart, {...action.payload, qty: 1}]
+            }
+        }
+        case 'ADD_TO_WISHLIST':{
+            return {
+                ...state, wishlist: [...state.wishlist, {...action.payload, qty: 1}]
+            }
+        }
```

Comment:
here you can do an initial check if the item is already in wishlist by comparing the payload's id with the items present in current state of wishlist. so that if item already exists you don't add new item for similar product.
you can do this check in product-cart page of here in this case.